packages/twenty-front/src/modules/object-record/record-index/options/hooks/useExportTableData.ts

```diff
@@ -0,0 +1,157 @@
+import { useEffect, useState } from 'react';
+import { json2csv } from 'json-2-csv';
+import { useRecoilValue } from 'recoil';
+
+import { useFindManyRecords } from '@/object-record/hooks/useFindManyRecords';
+import { FieldMetadata } from '@/object-record/record-field/types/FieldMetadata';
+import { useRecordTableStates } from '@/object-record/record-table/hooks/internal/useRecordTableStates';
+import { ColumnDefinition } from '@/object-record/record-table/types/ColumnDefinition';
+
+import { useFindManyParams } from '../../hooks/useLoadRecordIndexTable';
+
+export const sleep = (ms: number) =>
+  new Promise((resolve) => setTimeout(resolve, ms));
+
+export const download = (blob: Blob, filename: string) => {
+  const url = URL.createObjectURL(blob);
+  const link = document.createElement('a');
+  link.href = url;
+  link.setAttribute('download', filename);
+  document.body.appendChild(link);
+  link.click();
+  link.parentNode?.removeChild(link);
+};
+
+type GenerateExportOptions = {
+  columns: ColumnDefinition<FieldMetadata>[];
+  rows: object[];
+};
+
+type GenerateExport = (data: GenerateExportOptions) => string;
+
+export const generateCsv: GenerateExport = ({
+  columns,
+  rows,
+}: GenerateExportOptions): string => {
+  const keys = columns.flatMap((col) => {
+    const column = {
+      field: col.metadata.fieldName,
+      title: col.label,
+    };
+
+    const objectColumnData = rows.find((r) => {
```

Comment:
Could we add more explicit naming here ? 

- `r` => `row`
- `val` => `fieldValue`
- `const isObject = val && typeof val === 'object' && !Array.isArray(val); return isObject;`

Reply:
We should be able to push all of this into the dependency now that https://github.com/mrodrig/json-2-csv/issues/247 has landed